### ISSUE TITLE
Planemo test is ignoring `--failed`

### ISSUE THREAD

**Issue by bernt-matthias** [galaxyproject/planemo#1316]
planemo, version 0.75.2

**Comment by wm75** [comment]
yeah, #1352 is another case where an option became non-functional with that refactoring.
@simonbray rescued `--update_test_data` via https://github.com/galaxyproject/planemo/pull/1232/files#r885840184, but `--failed` and `--job_output_files` that also got handled through `run_in_config` were less lucky :(

**Comment by wm75** [comment]
Since an aspect of `--skip-venv` handling was also part of `run_in_config`. I am now wondering whether the broken behavior of that option that I mentioned in https://github.com/galaxyproject/planemo/issues/1343#issuecomment-1384194149 is also related.

**Comment by mvdbeek** [comment]
To start, **i** have added `--update_test_data` back in, for the sake of making progress with important work. Otherwise we'd still be stuck with 22.01. It was a deliberate choice to not keep this, and I'd rather we delete it for all the problem it has that i don't think are solvable, but y'all made the case that you want it, so that's ok. `--skip-venv` is such a niche flag that only bloats the options and makes reasoning about what people are doing with planemo more complicated, i hope we can just remove this (and maybe add this back when it really makes sense, that is when we have fully packaged galaxy).

The other flags are reasonable, and we can add them back, but this time with tests.

**Comment by wm75** [comment]
Cool! My own ordering by importance of these options is very similar:
`--update_test_data` == `--failed` > `--job_output_files` > `--skip-venv`
for the last, I don't have any actual use and would not mind if it was removed permanently.
For `--job_output_files`, there are workarounds like `--update_test_data` and git restoring what you want to keep unaltered, or `--no_cleanup` and taking files out of the job's work dir directly, but the option simplifies things.

**Comment by bernt-matthias** [comment]
fixed in https://github.com/galaxyproject/planemo/pull/1653 .. thanks @mvdbeek
